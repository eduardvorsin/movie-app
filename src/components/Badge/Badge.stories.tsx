import type { Meta, StoryObj } from '@storybook/react';
import Badge from './Badge';

const meta: Meta<typeof Badge> = {
	title: 'components/Badge',
	tags: ['autodocs'],
	argTypes: {
		appearance: {
			description: 'Appearance for badge',
		},
		className: {
			description: 'Additional classes for badge'
		},
		children: {
			description: 'Badge Contents',
		},
		testId: {
			description: 'Id for testing the component',
		}
	},
	component: Badge,
	parameters: {
		docs: {
			description: {
				component: 'Badges are employed to exhibit the condition of an object or the outcome of a completed action.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Badge>;

export const Default: Story = {
	args: {
		appearance: 'default',
		children: 'Default'
	},
	parameters: {
		docs: {
			description: {
				story: 'It is used to display objects, actions, or states that do not require special attention or do not have an explicit status',
			},
		},
	},
};

export const Info: Story = {
	args: {
		appearance: 'info',
		children: 'Info'
	},
	parameters: {
		docs: {
			description: {
				story: 'Used to send important or contextual messages to the user',
			},
		},
	},
};

export const Sucsess: Story = {
	args: {
		appearance: 'sucsess',
		children: 'Sucsess'
	},
	parameters: {
		docs: {
			description: {
				story: 'it is used to explicitly indicate successfully completed actions, completed tasks, or positive results',
			},
		},
	},
};

export const Warning: Story = {
	args: {
		appearance: 'warning',
		children: 'Warning'
	},
	parameters: {
		docs: {
			description: {
				story: 'It is used to highlight an object, action, or situation that requires special attention or may cause some risks',
			},
		},
	},
};

export const Danger: Story = {
	args: {
		appearance: 'danger',
		children: 'Danger'
	},
	parameters: {
		docs: {
			description: {
				story: 'It is used to explicitly designate situations, objects, or actions that pose a significant risk or negative consequences',
			},
		},
	},
};

const appearances = ['default', 'info', 'sucsess', 'warning', 'danger'] as const;

export const All: Story = {
	render: () => {
		return (
			<div
				className='flex flex-wrap items-start'
			>
				{appearances.map((appearance) => (
					<Badge
						className='mr-3 last:mr-0 mb-3 last:mb-0'
						key={appearance}
						appearance={appearance}
					>
						{`${appearance[0].toUpperCase()}${appearance.slice(1)}`}
					</Badge>
				))}
			</div>
		)
	},
	parameters: {
		docs: {
			description: {
				story: 'Here are all possible variants of badges by color',
			},
		},
	},
};