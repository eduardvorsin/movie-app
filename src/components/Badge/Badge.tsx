import { ReactNode } from "react";

type Props = {
	className?: string,
	testId?: string,
	appearance?: 'default' | 'info' | 'sucsess' | 'warning' | 'danger',
	children: ReactNode
};

const appearanceTypes = {
	default: 'bg-neutral-300',
	info: 'bg-blue-200',
	sucsess: 'bg-green-200',
	warning: 'bg-yellow-200',
	danger: 'bg-red-200',
} as const;

export default function Badge({
	className,
	children,
	testId,
	appearance = 'default',
}: Props) {
	const classes = [
		'inline-flex rounded-5 text-75 text-center leading-1 py-0.5 px-2',
		appearanceTypes[appearance],
		className,
	].join(' ');
	return (
		<span
			className={classes}
			data-testid={testId}
		>
			{children}
		</span>
	);
}