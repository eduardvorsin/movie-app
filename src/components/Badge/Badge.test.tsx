import { render, screen } from '@testing-library/react';
import Badge from './Badge';

describe('Badge tests', () => {
	it('is rendered correctly', () => {
		render(<Badge>Badge</Badge>);

		expect(screen.getByText<HTMLSpanElement>(/badge/i)).toBeInTheDocument();
	});

	it('is a basic snapshot', () => {
		render(<Badge>Badge</Badge>);

		expect(screen.getByText<HTMLSpanElement>(/badge/i)).toMatchSnapshot();
	});

	it('is a snapshot with the appearance type "default"', () => {
		render(<Badge appearance='default'>Badge</Badge>);

		expect(screen.getByText<HTMLSpanElement>(/badge/i)).toMatchSnapshot();
	});

	it('is a snapshot with the appearance type "info"', () => {
		render(<Badge appearance='info'>Badge</Badge>);

		expect(screen.getByText<HTMLSpanElement>(/badge/i)).toMatchSnapshot();
	});

	it('is a snapshot with the appearance type "success"', () => {
		render(<Badge appearance='success'>Badge</Badge>);

		expect(screen.getByText<HTMLSpanElement>(/badge/i)).toMatchSnapshot();
	});

	it('is a snapshot with the appearance type "warning"', () => {
		render(<Badge appearance='warning'>Badge</Badge>);

		expect(screen.getByText<HTMLSpanElement>(/badge/i)).toMatchSnapshot();
	});

	it('is a snapshot with the appearance type "danger"', () => {
		render(<Badge appearance='danger'>Badge</Badge>);

		expect(screen.getByText<HTMLSpanElement>(/badge/i)).toMatchSnapshot();
	});
});

