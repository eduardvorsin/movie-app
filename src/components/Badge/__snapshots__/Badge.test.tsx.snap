// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge tests is a basic snapshot 1`] = `
<span
  class="inline-flex rounded-5 text-75 text-center leading-1 py-0.5 px-2 bg-neutral-300 "
>
  Badge
</span>
`;

exports[`Badge tests is a snapshot with the appearance type "danger" 1`] = `
<span
  class="inline-flex rounded-5 text-75 text-center leading-1 py-0.5 px-2 bg-red-200 "
>
  Badge
</span>
`;

exports[`Badge tests is a snapshot with the appearance type "default" 1`] = `
<span
  class="inline-flex rounded-5 text-75 text-center leading-1 py-0.5 px-2 bg-neutral-300 "
>
  Badge
</span>
`;

exports[`Badge tests is a snapshot with the appearance type "info" 1`] = `
<span
  class="inline-flex rounded-5 text-75 text-center leading-1 py-0.5 px-2 bg-blue-200 "
>
  Badge
</span>
`;

exports[`Badge tests is a snapshot with the appearance type "success" 1`] = `
<span
  class="inline-flex rounded-5 text-75 text-center leading-1 py-0.5 px-2 bg-green-200 "
>
  Badge
</span>
`;

exports[`Badge tests is a snapshot with the appearance type "warning" 1`] = `
<span
  class="inline-flex rounded-5 text-75 text-center leading-1 py-0.5 px-2 bg-yellow-200 "
>
  Badge
</span>
`;
